src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -6,48 +6,49 @@
 
 public class ProductBasket {
 
-    private final List<Product> products = new LinkedList<>();
+    private final TreeMap<String, Product> products = new TreeMap<>();
```

Comment:
не забываем про полиморфизм
```suggestion
    private final Map<String, Product> products = new TreeMap<>();
```

Reply:
значение должно быть - список продуктов, а не один продукт